lib/Finance/Quote/CurrencyRates/AlphaVantage.pm

```diff
@@ -78,12 +78,13 @@ sub multipliers
     sleep (20) if (($try_cnt < 5) && ($json_data->{'Note'}));
   } while (($try_cnt < 5) && ($json_data->{'Note'}));
 
-  if( !$json_data->{'Realtime Currency Exchange Rate'} ) {
+  if( !$json_data->{'Time Series FX (Daily)'} ) {
     ### No data in JSON
 	  $rate = 0.0;
   } else {
+  my $target = substr($json_data->{'Meta Data'}->{'5. Last Refreshed'}, 0, 10);
```

Comment:
```
use List::Util qw( maxstr );
```
```suggestion
  my $target = maxstr(keys %{$json_data->{'Time Series FX (Daily)'}});
```

Reply:
This doesn't work for me. No result from `maxstr(keys %{$json_data->{'Time Series FX (Daily)'}})`

Reply:
Why not `my $target = (sort { $b cmp $a } keys %{ $json_data->{'Time Series FX (Daily)'} } )[0];`
I'm NOT a perl dev. I've just found similar question question on [stackoverflow](https://stackoverflow.com/questions/47916191/decode-json-and-return-first-key-in-hash)